src/lib/datepicker/datepicker-input.ts

```diff
@@ -116,22 +114,25 @@ export class MatDatepickerInput<D> implements AfterContentInit, ControlValueAcce
   }
   _dateFilter: (date: D | null) => boolean;
 
+  /** Model for current selected date */
+  _selectionModel: MatDateSelectionModel<D>;
+
   /** The value of the input. */
   @Input()
-  get value(): D | null { return this._value; }
+  get value(): D | null { return <D|null>this._selectionModel.selected; }
   set value(value: D | null) {
     value = this._dateAdapter.deserialize(value);
-    this._lastValueValid = !value || this._dateAdapter.isValid(value);
```

Comment:
This needs to be checked before we overwrite invalid values with null on the next line. I also think we probably need to delegate this validity checking to either the subclass or the selection model, the base class doesn't know if its dealing with a date or a date range